src/ai/rewardfunction/AttackHolderRewardFunction.java

```diff
@@ -0,0 +1,40 @@
+package ai.rewardfunction;
+
+import java.util.List;
+
+import rts.GameState;
+import rts.PhysicalGameState;
+import rts.TraceEntry;
+import rts.UnitAction;
+import rts.units.Unit;
+import util.Pair;
+
+public class AttackHolderRewardFunction extends RewardFunctionInterface {

```

Comment:
Based on your description, could you change the name from `AttackHolderRewardFunction` to `AttackFlagHolderRewardFunction`